File: 프로그래머스/1/42748. K번째수/K번째수.py
def solution(array, commands):
    answer = []
    
    i=0
    j=0
    k=0
    
    for x in range (len(commands)) :
        i = commands[x][0]
        j = commands[x][1]
        k = commands[x][2]
        
        answerarr=sorted(array[i-1:j])
        
        answer.append(answerarr[k-1])
    
    
    
    return answer

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.* ;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        List<Integer>answerlist = new ArrayList<>();
        
        for (int i = 0 ;i < commands.length; i++)
        {
            int start = commands[i][0]-1;
            int end = commands[i][1]-1;
            int k = commands[i][2]-1;
            
            List<Integer> comparelist = new ArrayList<>();
            
            for(int j = start;j<end+1;j++)
            {
                comparelist.add(array[j]);
            }
            Collections.sort(comparelist);
            
            answerlist.add(comparelist.get(k));
            
        }
        int [] answer = new int[answerlist.size()];
        for(int i=0; i<answerlist.size(); i++)
        {
            answer[i] = answerlist.get(i);
        }
        return answer;
    }
}